tests/widgets_table.rs

```diff
@@ -747,11 +747,94 @@ fn widgets_table_enable_always_highlight_spacing() {
 }
 
 #[test]
-fn widgets_table_can_have_elements_styled_individually() {
+fn widgets_table_can_have_spacing_in_all_columns() {
```

Comment:
It's tempting to continue with writing these integration style tests, but they have a tendency to be a bit verbose compared to an equivalent unit test. E.g.:

```rust
let buffer = Buffer::empty(Rect::new(0, 0, 15, 7));
let base_table = Table::new(vec![Row::new(vec!["C1", "C2", "C3"])]; 4)
  .header(Row::new(vec!["H1", "H2", "H3"])
  .block(Block::new().borders(Borders::ALL))
  ...

let mut state = TableState::default();
base_table.clone()
  .highlight_spacing(true)
  .render_stateful_widget(buffer.area, buffer, &mut state);
assert_buffer_eq!(
  buffer,
  buffer::with_lines(vec![
    "┌─────────────┐",
    "│H1 H2 H3     │",
    "│             │",
    "│C1 C2 C3     │",
    "│C1 C2 C3     │",
    "│C1 C2 C3     │",
    "└─────────────┘",
  ]),
);

state.select(Some(TableSelection::Row(0)));
base_table.clone()
  .highlight_spacing(true).
  .render_stateful_widget(buffer.area, buffer, &mut state);
assert_buffer_eq!(
  buffer,
  buffer::with_lines(vec![
    "┌─────────────┐",
    "│   H1 H2 H3  │",
    "│             │",
    "│>> C1 C2 C3  │",
    "│   C1 C2 C3  │",
    "│   C1 C2 C3  │",
    "└─────────────┘",
  ]),
);
```

For testing highlight styles, you can assign the expected buffer to a variable and call `expected.set_style(Rect::new(...))`). See https://github.com/ratatui-org/ratatui/blob/main/src/widgets/tabs.rs#L212 for an example of this.

Note: this is a subjective comment rather than one which blocks this in any way. It would be nicer for the long term health to head in this direction. It's possible that this is more work than you have time available, so I'd understand if it wasn't a priority.